packages/authgear-web/src/container.ts

```diff
@@ -270,26 +272,55 @@ export class WebContainer {
   }
 
   /**
-   * Start authorization by opening authorize page
-   *
-   * @param options - authorize options
+   * Start authorization by redirecting to the authorization endpoint.
    */
   async startAuthorization(options: AuthorizeOptions): Promise<void> {
     const authorizeEndpoint = await this.authorizeEndpoint(options);
     window.location.href = authorizeEndpoint;
   }
 
   /**
-   * Finish authorization
+   * Start reauthentication by redirecting to the authorization endpoint.
+   */
+  async startReauthentication(options: ReauthenticateOptions): Promise<void> {
+    await this.baseContainer._refreshIDToken();
+
+    const idToken = this.getIDTokenHint();
+    if (idToken == null || !this.canReauthenticate()) {
+      throw new Error(
+        "You can only trigger reauthentication when canReauthenticate() returns true"
```

Comment:
If we throw error here, do we expect developer should check `canReauthenticate` before calling reauthentication? If this is the case, seems we need some way to ensure `canReauthenticate` is up to date? e.g. User originally doesn't have mfa, and added mfa in the setting page later.